src/core/settings.c

```diff
@@ -1087,3 +1087,61 @@ QuicSettingsDumpNew(
         }
     }
 }
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicSettingsGetParam(
+    _In_ const QUIC_SETTINGS* IncomingSettings,
+    _Inout_ uint32_t* OutgoingSize,
+    _Out_writes_bytes_opt_(*OutgoingSize)
+        QUIC_SETTINGS* OutgoingSettings
+    )
+{
+    uint32_t MinimumSettingsSize = (uint32_t)FIELD_OFFSET(QUIC_SETTINGS, DesiredVersionsList);
+
+    if (*OutgoingSize < MinimumSettingsSize) {
+        *OutgoingSize = MinimumSettingsSize;
+        return QUIC_STATUS_BUFFER_TOO_SMALL;
+    }
+
+    if (*OutgoingSize > sizeof(QUIC_SETTINGS)) {
```

Comment:
You want to break if a new caller calls an old dll. That's the only scenario this would break, and since other things might break silently we really should want that to explicitly.

Reply:
Fixed